pkg/controllers/plugin/metrics.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+)
+
+const (
+	metricResultSuccess = "success"
+	metricResultError   = "error"
+
+	metricReasonTemplateFailed = "template_failed"
+	metricReasonDiffFailed     = "diff_failed"
+	metricReasonUpgradeFailed  = "upgrade_failed"
+)
+
+var (
+	pluginReconcileTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "greenhouse_plugin_reconcile_total",
+		},
+		[]string{"pluginDefinition", "clusterName", "plugin", "organization", "result", "reason"})
+)
+
+func init() {
+	metrics.Registry.MustRegister(pluginReconcileTotal)
+}
+
+func updateMetrics(plugin *greenhousev1alpha1.Plugin) {
+	result := metricResultSuccess
+	reason := ""
+
+	for _, condition := range plugin.Status.Conditions {
+		if condition.IsUnknown() {
+			continue
+		}
+
+		switch condition.Type {
+		case greenhousev1alpha1.HelmReconcileFailedCondition:
+			if condition.IsTrue() {
+				result = metricResultError
+				reason = metricReasonTemplateFailed
+			}
+		case greenhousev1alpha1.HelmDriftDetectedCondition:
+			if condition.IsTrue() {
+				result = metricResultError
+				reason = metricReasonDiffFailed
+			}
+		case greenhousev1alpha1.StatusUpToDateCondition:
+			if condition.IsFalse() {
+				result = metricResultError
+				reason = metricReasonUpgradeFailed
+			}
```

Comment:
These are not reflecting errors in reconciling the helm release.

Reply:
Done